From fad26f7617ebd04905243da7c6c73ef61b11c647 Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Thu, 16 Nov 2023 17:21:20 +0100
Subject: [PATCH] arm64: dts: st: add UART console on stm32mp257f-dk boards

Add USART2 to support console on stm32mp257f-dk boards.

Change-Id: I22c225abf89a04913d411ac4be6248bf6c4c202d
Signed-off-by: Olivier BIDEAU <olivier.bideau@st.com>
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index 14f1d187e194..460a75f33075 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -18,6 +18,7 @@ / {
 	compatible = "st,stm32mp257f-dk", "st,stm32mp257";
 
 	aliases {
+		serial0 = &usart2;
 	};
 
 	chosen {
@@ -54,3 +55,13 @@ &sdmmc1 {
 	vqmmc-supply = <&scmi_vddio1>;
 	status = "okay";
 };
+
+&usart2 {
+	pinctrl-names = "default", "idle", "sleep";
+	pinctrl-0 = <&usart2_pins_a>;
+	pinctrl-1 = <&usart2_idle_pins_a>;
+	pinctrl-2 = <&usart2_sleep_pins_a>;
+	/delete-property/dmas;
+	/delete-property/dma-names;
+	status = "okay";
+};
-- 
2.39.5

